Refactor guides section + promote HTTP over GIT (#1338)
huggingface/huggingface_hub: docs/source/guides/repository.mdx

# Create and manage a repository

The Huggingface Hub is a collection of git repositories. [Git](https://git-scm.com/) is a tool vastly used in software
development to easily version projects when working collaboratively. This guide will show you how to interact with the
repositories on the Hub, especially:

- Create and delete a repository.
- Manage branches and tags. 
- Rename your repository.
- Update your repository visibility.
- Manage a local copy of your repository.

<Tip warning={true}>

> If you are used to software engineering, working with platforms such as Gitlab/Github/Bitbucket, your first instinct
> might be to use `git` CLI to clone your repo (`git clone`), commit changes (`git add, git commit`) and push them
> (`git push`). This is still valid when using the Huggingface Hub. However, software engineering and machine learning do
> not share the same requirements and workflows. The `huggingface_hub` library has been implemented to address most of the
> ML use cases in a more efficient way. You can read our [Git vs HTTP paradigm](../concepts/git_vs_http) explanation page
> for more details.

Reviewer: Briefly clarify what we mean by saying we address ML use cases more efficiently.

```suggestion
If you are used to working with platforms such as Gitlab/Github/Bitbucket, your first instinct
might be to use `git` CLI to clone your repo (`git clone`), commit changes (`git add, git commit`) and push them
(`git push`). This is still valid when using the Hugging Face Hub. However, software engineering and machine learning do
not share the same requirements and workflows. In ML, it may be more efficient to use HTTP, so you don't have to maintain a local folder with massive model weights. You can read our [Git vs HTTP paradigm](../concepts/git_vs_http) explanation page
for more details.
```

</Tip>

If you want to create and manage a repository on the Hub, your machine must be logged in. If you are not, please refer to
[this section](../quick-start#login). In the rest of this guide, we will assume that your machine is logged in.

> ## Repo creation and deletion

Reviewer: It'd be nice to add a sentence or two after this header to describe what this section is about so users can quickly skim it. In general, its good to not have an empty section header :)

### Create a repository

> Create an empty repository with [`create_repo`] and give it a name with the `repo_id` parameter. The `repo_id` is your namespace followed by the repository name: `{username_or_org}/{repo_name}`.

Reviewer: ```suggestion
Create an empty repository with [`create_repo`] and give it a name with the `repo_id` parameter. The `repo_id` is your namespace followed by the repository name: `(username_or_org/repo_name)`.
```

Author: Let's even remove the parenthesis
```suggestion
Create an empty repository with [`create_repo`] and give it a name with the `repo_id` parameter. The `repo_id` is your namespace followed by the repository name: `username_or_org/repo_name`.
```

'https://huggingface.co/datasets/lysandre/test-dataset'
```

When you create a repository, you can set your repository visibility with the `private` parameter. For example, if you want to create a private repository:

```py
>>> from huggingface_hub import create_repo
>>> create_repo("lysandre/test-private", private=True)
```

If you want to change the repository visibility at a later time, you can use the [`update_repo_visibility`] function.

### Delete a repository

Delete a repository with [`delete_repo`]. Make sure you want to delete a repository because this is an irreversible process!

Specify the `repo_id` of the repository you want to delete:

```py
>>> delete_repo(repo_id="lysandre/my-corrupted-dataset", repo_type="dataset")
```

## Upload and download files

Now that you have created your repository, you are interested in pushing changes to it and downloading files from it.

These 2 topics deserve their own guides. Please refer to the [upload](./upload) and the [download](./download) guides
to learn how to use your repository.

## Branches and tags

Git repositories often make use of branches to store different versions of a same repository.

> Tags can also be use to flag a specific state of your repository, for example when releasing a version.

Reviewer: ```suggestion
Tags can also be used to flag a specific state of your repository, for example, when releasing a version.
```

More generally, branches and tags are referred as [git references](https://git-scm.com/book/en/v2/Git-Internals-Git-References).

### Create branches and tags

You can create new branch and tags using [`create_branch`] and [`create_tag`]:

```py
>>> from huggingface_hub import create_branch, create_tag

# Create a branch on a Space repo from `main` branch
>>> create_branch("Matthijs/speecht5-tts-demo", repo_type="space", branch="handle-dog-speaker")

# Create a tag on a Dataset repo from `v0.1-release` branch
>>> create_branch("bigcode/the-stack", repo_type="dataset", revision="v0.1-release", tag="v0.1.1", tag_message="Bump release version.")
```

> See also [`delete_branch`] and [`delete_tag`].

Reviewer: ```suggestion
You can use the [`delete_branch`] and [`delete_tag`] functions in the same way to delete a branch or a tag.
```

### List all branches and tags

You can also list the existing git refs from a repository using [`list_repo_refs`]:

```py
>>> from huggingface_hub import list_repo_refs
>>> api.list_repo_refs("bigcode/the-stack", repo_type="dataset")
GitRefs(
   branches=[
         GitRefInfo(name='main', ref='refs/heads/main', target_commit='18edc1591d9ce72aa82f56c4431b3c969b210ae3'),
         GitRefInfo(name='v1.1.a1', ref='refs/heads/v1.1.a1', target_commit='f9826b862d1567f3822d3d25649b0d6d22ace714')
   ],
   converts=[],
   tags=[
         GitRefInfo(name='v1.0', ref='refs/tags/v1.0', target_commit='c37a8cd1e382064d8aced5e05543c5f7753834da')
   ]
)
``` 

> ## Change repository configuration

Reviewer: I'd also add a sentence or two after this empty section header to describe what its about

### Update visibility

A repository can be public or private. A private repository is only visible to you or members of the organization in which the repository is located. Change a repository to private as shown in the following:

```py
>>> from huggingface_hub import update_repo_visibility
>>> update_repo_visibility(repo_id=repo_id, private=True)
```

### Rename your repository

You can rename your repository on the Hub using [`move_repo`]. Using this method, you can also move the repo from a user to
an organization. When doing so, there are a [few limitations](https://hf.co/docs/hub/repositories-settings#renaming-or-transferring-a-repo)

> that you should be aware of.

Reviewer: Just for a quick preview of what a limitation is

```suggestion
that you should be aware of. For example, you can't transfer your repo to another user.
```

# Create and manage a repository

The Hugging Face Hub is a collection of git repositories. [Git](https://git-scm.com/) is a widely used tool in software
development to easily version projects when working collaboratively. This guide will show you how to interact with the
repositories on the Hub, especially:

- Create and delete a repository.
- Manage branches and tags. 
- Rename your repository.
- Update your repository visibility.
- Manage a local copy of your repository.

<Tip warning={true}>

If you are used to working with platforms such as Gitlab/Github/Bitbucket, your first instinct
might be to use `git` CLI to clone your repo (`git clone`), commit changes (`git add, git commit`) and push them
(`git push`). This is still valid when using the Hugging Face Hub. However, software engineering and machine learning do

> not share the same requirements and workflows. In ML, it may be more efficient to use HTTP, so you don't have to maintain

Author: Yep completely! I'm updating this :)
